src/detect-tos.c

```diff
@@ -157,6 +161,9 @@ static DetectTosData *DetectTosParse(const char *arg, bool negate)
 
     return tosd;
 
+error_free:
+    if (str_ptr)
```

Comment:
```
** CID 1443023:  Null pointer dereferences  (REVERSE_INULL)
/src/detect-tos.c: 165 in DetectTosParse()


________________________________________________________________________________________________________
*** CID 1443023:  Null pointer dereferences  (REVERSE_INULL)
/src/detect-tos.c: 165 in DetectTosParse()
159         tosd->tos = (uint8_t)tos;
160         tosd->negated = negate;
161     
162         return tosd;
163     
164     error_free:
>>>     CID 1443023:  Null pointer dereferences  (REVERSE_INULL)
>>>     Null-checking "str_ptr" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.
165         if (str_ptr)
166             pcre_free_substring(str_ptr);
167     error:
168         return NULL;
169     }
170     
```